decentraland/components/presentational/Button.js

```diff
@@ -8,34 +8,56 @@ import {
   responsiveFontSize,
 } from "react-native-responsive-dimensions";
 
+// TODO: Change to NativeBase button component
+// https://github.com/tasitlabs/tasit/issues/204
 export default function Button(props) {
+  let { title, onPress, disabled } = props;
+  if (!disabled) disabled = false;
+  if (!onPress) onPress = () => {};
```

Comment:
Are we sure we want this to be an optional prop that gets defaulted?

Reply:
Which prop?

Reply:
If I say "this", I usually mean the exact line I'm commenting on, which is always the lowest one in the preview on the GitHub UI.

So in this case, I mean the `onPress` one

If it's a broader concept, I try to make sure to comment on the lowest relevant line, so that any lines above you that need to see for context will show up.

Reply:
Oh, ok. Thank you to clarify that! The UI is highlighting three lines. Got it now.
No strong opinion, doing another look maybe it isn't a good option. Maybe make more sense having `onPress` as optional if `disabled` is true. I've looked that it is possible since `Proptype` supports conditional requirement but I'm not sure if it worth increasing the complexity that way.
So, should I back `onPress` as a required prop?

Reply:
I think the way to go is to make `onPress` required even for disabled buttons, and then internally in the button component have the function that passed the `onPress` prop into the handler for the button do an if on whether disabled is true or not and don't call the `onPress` function that was passed in if disabled is true.

Reply:
That way the user can either pass in the real function that would be called if the button were enabled (which leaves us open for logic that just toggles the disabled state but keeps everything else the same) or pass in a mock function that doesn't do anything

Reply:
I agree that conditional requirement of props is too much complexity for now (and maybe is always a code smell), so I like that the strategy above gets around that

Reply:
> I think the way to go is to make `onPress` required even for disabled buttons, and then internally in the button component have the function that passed the `onPress` prop into the handler for the button do an if on whether disabled is true or not and don't call the `onPress` function that was passed in if disabled is true.

Sorry @pcowgill but I'm not sure if I've got your message, please correct me if I'm wrong:
1. `onPress` as required
2. Use `onPress` prop on the button if `disabled` != true.

Is it?

Reply:
> Sorry @pcowgill but I'm not sure if I've got your message, please correct me if I'm wrong:
>    1. onPress as required
>    2. Use onPress prop on the button if disabled != true.
>
> Is it?

Yep! For 2, I might word it as still use a function wrapping onPress on the button regardless, but only execute the onPress code within that function if `diabled` != `true`

Reply:
Done.

Reply:
Thanks! Resolving